Sources/ClientRuntime/Networking/Http/URLSession/URLSessionConfiguration+HTTPClientConfiguration.swift

```diff
@@ -0,0 +1,23 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#if os(iOS) || os(macOS) || os(watchOS) || os(tvOS) || os(visionOS)
```

Comment:
Enabled on Mac because:
- Because of weirdness using Foundation thread objects in simulators, macOS is the only platform where this client can be tested correctly on CI.
- It will not be the default on Mac, but it will be an option for Mac platform apps.  (I would prefer URLSession for a Mac app that does low-volume AWS requests in the background, versus the CRT client for an app that does high-volume AWS requests in the foreground.)

I am not providing a way to enable the Foundation client on Mac at this time.  That will be provided with SRA Plugins work, which starts soon.